packages/web-components/@groupby/elements-products/src/products-sayt.ts

```diff
@@ -40,6 +57,18 @@ export default class ProductsSayt extends ProductsBase {
     super.disconnectedCallback();
 
     window.removeEventListener(SAYT_PRODUCTS_RESPONSE, this.setProductsFromEvent);
+    window.removeEventListener(this.cacheResponseEventName, this.setProductsFromCacheEvent);
+  }
+
+  /**
+   * Receives an event for populating initial data.
+   * Intended to be used on mount of this component.
+   *
+   * @param event The event object.
+   */
+  setProductsFromCacheEvent(event: CustomEvent<CacheResponsePayload>): void {
+    const data = event.detail.data || {};
```

Comment:
@DanielMoniz-GBI: I would normally suggest the refactor below, but it does change the behaviour slightly (defaulting only on undefined rather than any falsy value). Would that be acceptable in this case?  

```suggestion
    const { data = {} } = event.detail;
```

@ssbelle If you copied this code from autocomplete, then ignore this suggestion for consistency.

Reply:
I'm assuming we'd want to default on `undefined` and maybe `null`. We probably don't want to default on any other value.

Reply:
@jmbtutor I did copy this code essentially from the autocomplete example here: https://github.com/groupby/elements-view/pull/45/files

Reply:
Did you want me to change this? Or leave the same, as it is the same as autocomplete @jmbtutor @DanielMoniz-GBI

Reply:
leave